grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -2395,6 +2395,7 @@ class StockMovementService {
                 shipmentItems.each { ShipmentItem shipmentItem ->
                     shipment.addToShipmentItems(shipmentItem)
                 }
+                shipment.save(flush: true)
```

Comment:
And for the first case (`if`) it is not required?

Reply:
Oh, you're right. I didn't see that before 🙉